shared/util.go

```diff
@@ -268,13 +279,13 @@ func LogPath(path ...string) string {
 //     One of `file`, `symlink`, `directory`
 //   - `X-LXD-write`
 //     One of `overwrite`, `append`
-func ParseLXDFileHeaders(headers http.Header) (uid int64, gid int64, mode int, type_ string, write string) {
+func ParseLXDFileHeaders(headers http.Header) (*LXDFileHeaders, error) {
```

Comment:
Now that this can return an error we should ensure that invalid values in the headers return an error rather than being ignored.

Reply:
Fixed. I had originally avoided doing this since it changes the behavior of the API.